test/e2e/framework/framework.go

```diff
@@ -0,0 +1,274 @@
+
+    SkipNamespaceCreation    bool            // Whether to skip creating a namespace
+    Namespace                string          // Every test has a namespace at least unless creation is skipped
+    namespacesToDelete       []*v1.Namespace // Some tests have more than one.
+    NamespaceDeletionTimeout time.Duration
+
+    // To make sure that this framework cleans up after itself, no matter what,
+    // we install a Cleanup action before each test and clear it after.  If we
+    // should abort, the AfterSuite hook should run all Cleanup actions.
+    cleanupHandle            CleanupActionHandle
+
+    // configuration for framework's client
+    Options Options
+
+}
+
+// Options is a struct for managing test framework options.
+type Options struct {
+    ClientQPS    float32
+    ClientBurst  int
+    GroupVersion *schema.GroupVersion
+}
+
+// NewDefaultFramework makes a new framework and sets up a BeforeEach/AfterEach for
+// you (you can write additional before/after each functions).
+func NewDefaultFramework(baseName string) *Framework {
+    options := Options{
+        ClientQPS:   20,
+        ClientBurst: 50,
+    }
+    return NewFramework(baseName, options)
+}
+
+// NewFramework creates a test framework.
+func NewFramework(baseName string, options Options) *Framework {
+    f := &Framework{
+        BaseName:                 baseName,
+        Options:                  options,
+    }
+
+    ginkgo.BeforeEach(f.BeforeEach)
+    ginkgo.AfterEach(f.AfterEach)
+
+    return f
+}
+
+
+func (f *Framework) BeforeEach() {
+    // workaround for a bug in ginkgo.
+    // https://github.com/onsi/ginkgo/issues/222
+    f.cleanupHandle = AddCleanupAction(f.AfterEach)
+
+    ginkgo.By("Creating kubernetes clients")
+    for _, context := range TestContext.KubeContexts {
+        client := f.createKubernetesClient(context)
+        f.ClusterClients = append(f.ClusterClients, client)
+    }
+
+    if !f.SkipNamespaceCreation {
+        ginkgo.By(fmt.Sprintf("Building namespace api objects, basename %s", f.BaseName))
+
+        namespaceLabels := map[string]string {
+            "e2e-framework": f.BaseName,
+        }
+
+        for idx, clientSet := range f.ClusterClients {
+            switch idx {
+            case 0: // On the first cluster we let k8s generate a name for the namespace
+                namespace := generateNamespace(clientSet, f.BaseName, namespaceLabels)
+                f.Namespace = namespace.GetName()
+                f.UniqueName = namespace.GetName()
+            default: // On the other clusters we use the same name to make tracing easier
+                f.CreateNamespace(clientSet, f.Namespace, namespaceLabels)
+            }
+        }
+    } else {
+        f.UniqueName = string(uuid.NewUUID())
+    }
+
+}
+
+func (f *Framework) createKubernetesClient(context string) *kubeclientset.Clientset {
+
+    restConfig, _, err := loadConfig(TestContext.KubeConfig, context)
+    Expect(err).NotTo(HaveOccurred())
```

Comment:
We should return on error...

Reply:
no, this is for a test framework, at this point what I want is to make the test stop, because there was an error with the config file probably.

Reply:
But I wonder if may be in this case it would be worth to stop the whole test execution all together.

Reply:
yeah if an error failure stops the test then that's fine.